### PR TITLE
Android prebuild

### PR DESCRIPTION
**Pull request by mribbons**
Android prebuilds are built by default, as long as `ANDROID_HOME` is set.

See notes in https://github.com/socketsupply/socket/blob/android-prebuild/bin/build-android-runtime-libraries.sh

CI:
`ANDROID_HOME` seems to be set in all desktop variants.
Building Android for these slows things down too much, therefore we only build android if
`SSC_ANDROID_CI` is set.

### CODE REVIEW

File: src/cli/templates.hh [hunk 1]
```diff
@@ -1422,6 +1415,7 @@ version = 1.0.0
 
 
 [android]
+build_remove_path = dist/android/app/src/
```
Comment (chicoxyzzy) [inline src/cli/templates.hh:1418]: can we use the `settings["output"] + /android/app/src/` in cli.cc to not introduce extra config parameters?

Comment (heapwolf) [reply src/cli/templates.hh:1418]: +1 if this is possible, I do this in some other places

Comment (mribbons) [reply src/cli/templates.hh]: Seems to work fine, all done.

**Review by chicoxyzzy — APPROVED** [review]
I don't have an Android device to test on, but the code looks good to me

**Comment by chicoxyzzy** [comment]
hmm macOS tests are failing

**Comment by jwerle** [comment]
@mribbons can we close this one in favor of #234 ?

**Pull request closed without merging**